### ISSUE TITLE
Ensure that dataset.csv is checked right at the start of training

### ISSUE THREAD

**Issue by ant0nsc** [microsoft/InnerEye-DeepLearning#428]
For segmentation models: Errors in dataset.csv for the test set are only caught when the test set is actually used. 
Need to ensure that this happens right at the start of training.
@kh296 

[AB#3943](https://innereye.visualstudio.com/60ce1777-00d6-4015-82bc-488a0c00202f/_workitems/edit/3943)

**Comment by ant0nsc** [comment]
Use this dataset with the `DUmmyModel` to repro:
```
subject,filePath,channel,institutionId
1,train_and_test_data/id1_channel1.nii.gz,channel1,1
1,train_and_test_data/id1_channel1.nii.gz,channel2,1
1,train_and_test_data/id1_mask.nii.gz,mask,1
1,train_and_test_data/id1_region.nii.gz,region,1
1,train_and_test_data/id1_region.nii.gz,region_1,1
2,train_and_test_data/id2_channel1.nii.gz,channel1,2
2,train_and_test_data/id2_channel1.nii.gz,channel2,2
2,train_and_test_data/id2_mask.nii.gz,mask,2
2,train_and_test_data/id2_region.nii.gz,region,2
2,train_and_test_data/id2_region.nii.gz,region_1,2
3,train_and_test_data/id2_channel1.nii.gz,channel1,3
3,train_and_test_data/id2_channel1.nii.gz,channel2,3
3,train_and_test_data/id2_mask.nii.gz,mask,3
3,train_and_test_data/id2_region.nii.gz,region,3
3,train_and_test_data/id2_region.nii.gz,region_1,3
4,train_and_test_data/id2_channel1.nii.gz,channel1,3
4,train_and_test_data/id2_channel1.nii.gz,channel2,3
4,train_and_test_data/id2_mask.nii.gz,mask,3
4,train_and_test_data/id2_region.nii.gz,region,3
4,train_and_test_data/id2_region.nii.gz,region_1,3
5,train_and_test_data/id2_channel1.nii.gz,channel1,3
5,train_and_test_data/id2_channel1.nii.gz,channel2,3
5,train_and_test_data/id2_mask.nii.gz,mask,3
5,train_and_test_data/id2_region.nii.gz,region,3
5,train_and_test_data/id2_region.nii.gz,region_1,3
6,train_and_test_data/id2_channel1.nii.gz,channel1,3
6,train_and_test_data/id2_channel1.nii.gz,channel2,3
6,train_and_test_data/id2_mask.nii.gz,mask,3
6,train_and_test_data/id2_region.nii.gz,region,3
6,train_and_test_data/id2_region.nii.gz,region_1,3
7,train_and_test_data/id2_channel1.nii.gz,channel1,4
7,train_and_test_data/id2_channel1.nii.gz,channel2,4
7,train_and_test_data/id2_mask.nii.gz,mask,4
7,train_and_test_data/id2_region.nii.gz,region_1,4
```
(one line for patient 7 missing)

**Comment by ant0nsc** [comment]
Fixed by #447 . @kh296  FYI
